fda-model/ead/simulation.cs

```diff
@@ -438,6 +429,11 @@ public SimulationBuilder withPerformanceMetrics(Results mr)
                 _sim._results = mr;
                 return new SimulationBuilder(_sim);
             }
+            public SimulationBuilder withAdditionalThreshold(Threshold threshold)
+            {
+                _sim._results.Thresholds.AddThreshold(threshold);
```

Comment:
im not entirely sure, but you might need to check that thresholds is not null.

Reply:
Check if threshold is null or thresholds? 

We have tests passing with this structure, the simulation is being built with the additional threshold without hitting exceptions.